pds_doi_core/outputs/osti_web_parser.py

```diff
@@ -58,6 +58,46 @@ def response_get_parse_osti_xml(self,osti_response_text):
 
         return o_response_dict
 
+    def response_get_parse_osti_xml_multiple_records(self,osti_response_text):
+        """Function parse a response from a GET (query) or a PUT to the OSTI server (in XML query format) and return a list of dictionaries.
+           Specific fields are extracted from input.  Not all fields in XML are used."""
+
+        o_response_dicts = []
+
+        doc     = etree.fromstring(osti_response_text)
+        my_root = doc.getroottree()
+
+        # Trim down input to just fields we want.
+        element_record = 0
+        for element in my_root.iter():
+            if element.tag == 'record':
+                my_record = my_root.xpath(element.tag)[0]
```

Comment:
Can we re-use the method 'response_get_parse_osti_xml' ? It would need to be updated maybe as here you are parsing more field than above.

Another question, what is the difference between this method and one in the input sub-package ? I am getting lost... But there might be some duplication of code that we would like to avoid.

Reply:
Code has been reworked to be reused.

Reply:
Ok perfect, this made the osti object in the osti package not used anymore. I removed it.